clients/ts/signalr/src/ServerSentEventsTransport.ts

```diff
@@ -11,14 +11,20 @@ export class ServerSentEventsTransport implements ITransport {
     private readonly accessTokenFactory: () => string | Promise<string>;
     private readonly logger: ILogger;
     private readonly logMessageContent: boolean;
-    private eventSource: EventSource;
-    private url: string;
+    private eventSource?: EventSource;
+    private url?: string;
 
-    constructor(httpClient: HttpClient, accessTokenFactory: () => string | Promise<string>, logger: ILogger, logMessageContent: boolean) {
+    public onreceive: ((data: string | ArrayBuffer) => void) | null;
+    public onclose: ((error?: Error) => void) | null;
+
+    constructor(httpClient: HttpClient, accessTokenFactory: (() => string | Promise<string>) | undefined, logger: ILogger, logMessageContent: boolean) {
         this.httpClient = httpClient;
-        this.accessTokenFactory = accessTokenFactory || (() => null);
+        this.accessTokenFactory = accessTokenFactory || (() => "");
```

Comment:
I think this should be null and accessTokenFactory should be:

accessTokenFactory: (() => string | Promise<string> | null)

Reply:
accessTokenFactory is passed in from the end user, and the end user would never have a reason to return null from it.

The result from the factory is always checked to not be an empty string so it shouldn't matter.
Alternatively the code could be updated to not set a lambda at all.

Reply:
> Alternatively the code could be updated to not set a lambda at all.

Sure, lets do this then since we're still null checking it.